syntax/decode.go

```diff
@@ -16,6 +16,15 @@ func Unmarshal(data []byte, v interface{}) (int, error) {
 	return d.unmarshal(v)
 }
 
+// Unmarshaler is the interface implemented by types that can
+// unmarshal a TLS description of themselves.  Note that unlike the
+// JSON unmarshaler interface, it is not known a priori how much of
+// the input data will be consumed.  So the Unmarshaler must state
+// how much of the input data it consumed.
+type Unmarshaler interface {
+	UnmarshalTLS([]byte) (int, error)
+}
+
```

Comment:
I wonder if we should instead have this take bytes.Buffer, then you could examine Left instead. I guess that's my argument for this interface as a whole.

Reply:
I can see why that seems appealing, but I still like the `[]byte` approach better because (1) it's consistent with `syntax.Unmarshal()`, and (2) it's a slightly simpler interface for classes to implement.

Reply:
You could probably do a mixin for that, but ok